rest/client.go

```diff
@@ -179,8 +180,12 @@ func (c Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 
 	if v != nil {
 		// Try to unmarshal body into given type using streaming decoder.
-		if err := json.NewDecoder(resp.Body).Decode(&v); err != nil {
-			return nil, err
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, fmt.Errorf("error reading response body: %w", err)
+		}
```

Comment:
I think we already perform this in `CheckResponse`, which seems like the more appropriate place to check for json decoding errors.  Can we make the error returned from that method more verbose instead of making this section more complex?

Reply:
@AskAlice How things are going with this implementation, does it make sense?